doc/test-requirements-template.md

```diff
@@ -27,22 +27,50 @@ Write a few sentences or paragraphs describing the purpose and scope of the test
 
 ## Procedure
 
-* Test environment setup
+### Test environment setup
   * Description of procedure to configure ATE and DUT with pre-requisites making it possible to cover the intended paths and RPC's.
 
-* TestID-x.y.z - Name of subtest
+### TestID-x.y.z - Name of subtest
+  * Canonical OpenConfig Configuration
+
+An example OpenConfig configuration and/or RPC content should be specified here in YAML format.
+
+```yaml
+---
+openconfig-qos:
```

Comment:
I'm supportive of adding this to the README. It'd be great to ensure the paths that are mentioned later are covered by this example (or a set of examples).

I do have concerns about using YAML here before we have a formal specification written for it. We have done this before, with not great results -- when a pre-7951 standard was used in a manner that was not externally defined. If we are adopting YAML here, I'd like to see some efforts going toward defining a formal specification à la RFC7951 for this encoding, and having this published somewhere.

Reply:
Note that without a specification and some tooling support (which I can help to write if we have a specification) then you cannot validate this YAML.

Reply:
For validation, I wonder if we can take the approach of parsing, converting to JSON and running any desired  validations as JSON?  This may reduce the burden of writing yaml specific validations.  

YAML is a superset of JSON and I am sure I don't want to use any YAML feature which does not map to JSON.  

There is some precedent for this YAML->JSON approach  in go here: https://github.com/ghodss/yaml?tab=readme-ov-file.  The referenced blog post goes through their rationale of this approach.

Reply:
IMO, we shouldn't block the request for authors to write canonical OC in the README on validation.  But I do agree we ought add such validation.   It might take some time to get this validation up and running ad while that's happening we can be benefitting from README's that have (possibly model incorrect) canonical OC.   This is an incremental improvement on top of the flat list of OC paths  we currently have, which are verified.

The flat list of paths in ocpaths subsection of the readme markdown is validated here: [tools/validate_readme_spec](https://github.com/openconfig/featureprofiles/tree/main/tools/validate_readme_spec). I realize this is not at all the same thing as validating the OC content of a canonical config.   It's merely some an indication of incremental progress to making defining requirements better.  :)

Reply:
Note: In the future, the canonical OC content will be validated for
OC schema correctness and also replace the flat list of paths in the
"OpenConfig Path and RPC Coverage" heading.  

But until that validation is in place, we'll keep the flat list of OC paths because that is validated right now.

Reply:
SGTM.